autometa/common/utilities.py

```diff
@@ -298,14 +320,76 @@ def blockiter(fh, blocksize=65536):
                 yield block
                 block = fh.read(blocksize)
 
-    if type(fpath) != str:
+    if not isinstance(fpath, str):
         raise TypeError(type(fpath))
     if not os.path.exists(fpath):
         raise FileNotFoundError(fpath)
     fh = open(fpath, 'rb')
-    cksum = sha(blockiter(fh))
+    hash = md5sum(blockiter(fh))
     fh.close()
-    return cksum
+    return f'{hash} {os.path.basename(fpath)}\n'
+
+def read_checksum(fpath):
+    """Read checksum from provided checksum formatted `fpath`.
+
+    Note: See `write_checksum` for how a checksum file is generated.
+
+    Parameters
+    ----------
+    fpath : str
+        </path/to/file.md5>
+
+    Returns
+    -------
+    str
+        checksum retrieved from `fpath`.
+
+    Raises
+    -------
+    TypeError
+        Provided `fpath` was not a string.
+    FileNotFoundError
+        Provided `fpath` does not exist.
+
+    """
+    if not isinstance(fpath, str):
+        raise TypeError(type(fpath))
+    if not os.path.exists(fpath):
+        raise FileNotFoundError(fpath)
+    with open(fpath) as fh:
+        return fh.readline()
```

Comment:
This will return the checksum with a newline character at the end (if one is there). Is that what we want? Perhaps we should do `return fh.readline().rstrip()`?

Reply:
To keep consistency with other checksum files, I return the full line with special characters. For example, retrieving an md5 from NCBI would also contain the newline character, which would be reflected in the checksum calculation, so this is included.

```python
In [1]: write_checksum("VERSION", 'version.md5')

In [2]: !cat version.md5
3fda9608a77b3809141f75a6069b4abd VERSION

In [3]: with open('VERSION') as fh:
    ...:     line = fh.readline()

In [4]: line
Out[4]: '2.0a0\n'

In [5]: line.strip()
Out[5]: '2.0a0'

In [6]: with open('test','w') as fh:
    ...:     fh.write(line.strip())

In [7]: calc_checksum('test')
Out[7]: '907737a0287cdf5569eaf857e22f7e8c test\n'

In [8]: read_checksum('version.md5')
Out[8]: '3fda9608a77b3809141f75a6069b4abd VERSION\n'

In [9]: with open('test','w') as fh:
    ...:     fh.write(line)

In [10]: calc_checksum('test')
Out[10]: '3fda9608a77b3809141f75a6069b4abd test\n'

```

Reply:
OK, this makes sense.